### PR TITLE
Add OpenCL_jll dependence

### PR DESCRIPTION
**Pull request by ClaroHenrique**
This PR aims to add the OpenCL_jll dependency.
With that, the host machine dont need to have OpenCL installed.

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -3,8 +3,8 @@ uuid = "08131aa3-fb12-5dee-8b74-c09406e224a2"
 version = "0.8.1"
 
 [deps]
-Libdl = "8f399da3-3557-5675-b5ff-fb832c97cbdb"
 LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
+OpenCL_jll = "6cb37087-e8b6-5417-8430-1f242f1e46e4"
 Printf = "de0858da-6303-5e67-8744-51eddeeeb8d7"
```
Comment (juliohm) [inline Project.toml:9]: I think we also need to add a [compat] section

File: src/api.jl [hunk 2]
```diff
@@ -2,11 +2,9 @@ module api
 
 include("types.jl")
 
-const paths = Sys.isapple() ? String["/System/Library/Frameworks/OpenCL.framework"] : String[]
+import OpenCL_jll
```
Comment (juliohm) [inline src/api.jl:5]: It is great to see this hard-coded string gone

File: Project.toml (hunk 1 above)
Comment (ClaroHenrique) [reply Project.toml:9]: would be like this ?
```
[compat]
OpenCL_jll = "2022.9.23"
```

Comment (juliohm) [reply Project.toml:9]: Yes, I think that is right. Can you update the PR so that I can merge it?

**Comment by juliohm** [comment]
I am getting the following error with this PR:

```julia
julia> using OpenCL
julia> cl.platforms()
ERROR: CLError(code=-1001, CL_PLATFORM_NOT_FOUND_KHR)
```

**Comment by juliohm** [comment]
As we discussed on our call, this error is happening because of the missing OpenCL driver. Everything is working after a driver is manually installed.

**Pull request merged**